sled-agent/src/http_entrypoints.rs

```diff
@@ -630,3 +636,76 @@ async fn timesync_get(
     let sa = rqctx.context();
     Ok(HttpResponseOk(sa.timesync_get().await.map_err(|e| Error::from(e))?))
 }
+
+#[endpoint {
+    method = POST,
+    path = "/switch-ports",
+}]
+async fn uplink_ensure(
+    rqctx: RequestContext<SledAgent>,
+    body: TypedBody<SwitchPorts>,
+) -> Result<HttpResponseUpdatedNoContent, HttpError> {
+    let sa = rqctx.context();
+    sa.ensure_scrimlet_host_ports(body.into_inner().uplinks).await?;
+    Ok(HttpResponseUpdatedNoContent())
+}
+
+/// This API endpoint is only reading the local sled agent's view of the
+/// bootstore. The boostore is a distributed data store that is eventually
+/// consistent. Reads from individual nodes may not represent the latest state.
+#[endpoint {
+    method = GET,
+    path = "/network-bootstore-config",
+}]
+async fn read_network_bootstore_config_cache(
+    rqctx: RequestContext<SledAgent>,
+) -> Result<HttpResponseOk<EarlyNetworkConfig>, HttpError> {
+    let sa = rqctx.context();
+    let bs = sa.bootstore();
+
+    let config = bs.get_network_config().await.map_err(|e| {
+        HttpError::for_internal_error(format!("failed to get bootstore: {e}"))
+    })?;
+
+    let config = match config {
+        Some(config) => EarlyNetworkConfig::try_from(config).map_err(|e| {
+            HttpError::for_internal_error(format!(
+                "deserialize early network config: {e}"
+            ))
+        })?,
+        None => {
+            return Err(HttpError::for_unavail(
+                None,
+                "early network config does not exist yet".into(),
+            ));
+        }
+    };
+
+    Ok(HttpResponseOk(config))
+}
+
+#[endpoint {
+    method = PUT,
+    path = "/network-bootstore-config",
+}]
+async fn write_network_bootstore_config(
+    rqctx: RequestContext<SledAgent>,
+    body: TypedBody<EarlyNetworkConfig>,
+) -> Result<HttpResponseUpdatedNoContent, HttpError> {
+    let sa = rqctx.context();
+    let bs = sa.bootstore();
+    let config = body.into_inner();
+
+    //TODO is the bootstore checking the generation number, do we need to do
```

Comment:
Should probably remove this comment. This can also be done in a follow up to not extend build/merge times any more than necessary.

Reply:
Done